HHS/Head-Start-TTADP
# [TTAHUB-2942] Fix selected goal count

## Description of change

After looking into this bug it seems there was much more broken on this page than we first thought.

The following logic is now in place:

- The 'Select all' check box is respected per page (IE its only selected if all goals on that PAGE are selected or all goals on each page are selected)
- We now track checkbox selection based on the actual number of goals selected (additional rolled up goal ids are looked up at print time) .
- The select goals on ALL pages link only shows if you have all goals selected on that specific page (and don't currently have all selected).
![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/e189c720-bbaa-4ba3-938c-2c9fdfa1669a)
-  The pill of selected goals should show the number of goals selected on ALL page (so 10 on the first 2 on the second would show 12).
![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/945dbcc3-421a-4e06-a3b8-f586409c4e40)
-  There are two values shown in the alert, the first is the number selected on this page the second is all that will be selected if you click the link.
![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/c3fc92af-6dc5-4e68-9eca-e1e09ec82e5d)
- If no goals are selected we print all goals on that page ONLY.

## How to test

- Review the change.
- Test that the 'Select all' is respected per page.
- Check that we only show the select goals alert on all pages if all are selected on that page (and not already selected on all pages)
- Make sure we track selections globally (IE the first page might have all selected and the 3rd page only two goals).
- Test printing with all goals selected make sure rolled up goals are shown in the print.
- Test printing a page that has rolled up goals with nothing selected.
- Test printing with selected goal from various pages.
- Test printing with no goals selected.    

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2942


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
